### PR TITLE
Tony/prepare for new tests

### PR DESCRIPTION
**Pull request by it-ony**
# Problem

Tests should not check the content, but relay on unique id's per view.

# Solution

Introduce data-page-ids.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [n/a] Has the CHANGELOG been updated?
- [n/a] Has the README been updated?
- [n/a] Has the CONTRIBUTING doc been updated?
- [n/a] Has the RELEASE_GUIDELINES been updated?
- [n/a] Has the TESTING_GUIDELINES been updated?
- [n/a] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [n/a] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [n/a] Have any new automated tests been implemented or the existing ones changed?
- [n/a] Have any new manual tests been written down or the existing ones changed?
- [n/a] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: src/components/Router/CrossDeviceMobileRouter.tsx [hunk 1]
```diff
@@ -99,7 +99,9 @@ export default class CrossDeviceMobileRouter extends Component<
     // the cross device flow without running nginx
     const url = props.urls.sync_url
 
-    const roomId = window.location.pathname.substring(3) || props.options.roomId
+    // TODO: review this change
+
+    const roomId = props.options.roomId || window.location.pathname.substring(3)
```
Comment (it-ony) [inline src/components/Router/CrossDeviceMobileRouter.tsx:104]: When making the crossdevice working locally, it was always taking the input of the url in preference to the roomId passed as options. The or seems to be flipped. 

Please review @ethan605

File: src/components/crossDevice/CrossDeviceLink/index.js [hunk 2]
```diff
@@ -246,8 +246,11 @@ class CrossDeviceLinkUI extends Component {
     const mobileUrl = cross_device_url || hosted_sdk_url
     // This lets us test the cross device flow locally and on Surge.
     // We use the same location to test the same bundle as the desktop flow.
+
+    // TODO: review this change
+
     return process.env.MOBILE_URL === '/'
-      ? `${window.location.origin}?link_id=${this.linkId}`
+      ? `${window.location.origin}${window.location.pathname}?link_id=${this.linkId}`
```
Comment (it-ony) [inline src/components/crossDevice/CrossDeviceLink/index.js:253]: The mobile url was always generated without the pathname. Running locally under http://localhost/somedir/ ended up in a link generated http://localhost/?link_id skipping the path.

/cc @ethan605

Comment (stefaniacardenas) [reply src/components/crossDevice/CrossDeviceLink/index.js:253]: In which cases did you need to use the pathname locally? Anyways, the change looks good to me

**Review by stefaniacardenas — APPROVED** [review]
Looks good to me

File: src/components/crossDevice/CrossDeviceLink/index.js (hunk 2 above)
Comment (it-ony) [reply src/components/crossDevice/CrossDeviceLink/index.js:253]: For local testing without the need to rebuild the SDK. Before, we always assumed, that the cross origin url was served under http://host:port/.

File: src/components/Router/CrossDeviceMobileRouter.tsx (hunk 1 above)
Comment (stefaniacardenas) [reply src/components/Router/CrossDeviceMobileRouter.tsx:104]: As discussed, it should be safe to change this as we only need to fetch the room id from the URL when the roomId is not set in the store. So if roomId is null or undefined we can get it from the URL

**Pull request merged**